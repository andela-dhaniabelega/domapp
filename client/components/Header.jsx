import React from 'react';

const Header = () => ({
	render() {
		return (
			<div>
				<h1>This is the header</h1>
			</div>
		);
	}
});
export default Header;
