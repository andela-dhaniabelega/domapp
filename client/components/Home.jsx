import React from 'react';
import SignUpForm from './SignUpForm';

class Home extends React.Component {
	render(){
		return(
			<div>
				<h1>Home page</h1>
			</div>
		);
	}
}
export default Home;
