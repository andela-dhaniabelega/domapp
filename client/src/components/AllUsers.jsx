import React from 'react';
import { withRouter } from 'react-router-dom';
import _ from 'lodash';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import { confirmAlert } from 'react-confirm-alert';
import 'react-confirm-alert/src/react-confirm-alert.css';
import { allUsers } from '../actions/allUsers';
import { removeUser } from '../actions/removeUser';
import { searchUsers } from '../actions/searchUsers';
/**
 * @class AllUsers
 * @extends {React.Component}
 */
export class AllUsers extends React.Component {
	/**
	 * @description Creates and intializes objects
	 * @param  {object} props
	 * @return {void}
	 */
	constructor(props) {
		super(props);
		this.state = {
			searchQuery: '',
			searchResult: [],
			users: [],
			pagination: [],
			limit: 10,
			offset: 0
		};
		this.onChange = this.onChange.bind(this);
		this.handleDeleteUser = this.handleDeleteUser.bind(this);
		this.handlePageChange = this.handlePageChange.bind(this);
		this.handleSearch = this.handleSearch.bind(this);
	}
	/**
	 * @description Lifecycle Method
	 * Called when an instance of the component
	 * is created or inserted into the DOM
	 * @return {void}
	 */
	componentWillMount() {
		this.props.allUsers(this.state);
	}
	/**
	 * @description Lifecycle Method 
	 * Called before a mounted component receives props
	 * @param {object} [nextProps]
	 * @return {void}
	 */
	componentWillReceiveProps(nextProps) {
		this.setState({
			users: nextProps.users,
			pagination: nextProps.pagination
			// searchResult: nextProps.searchResult
		});
	}
	/**
	 * @description Allows user Interact with Input boxes
	 * @param  {object} e
	 * @return {void}
	 */
	onChange(e) {
		this.setState({ searchQuery: e.target.value });
	}
	/**
	 * @description Allows user navigate pages by changing offset
	 * @param  {object} page 
	 * @return {void}
	 */
	handlePageChange(page) {
		const selected = page.selected;
		const offset = Math.ceil(selected * this.state.limit);
		this.setState({ offset }, () => {
			this.props.allUsers(this.state);
		});
	}
	/**
	 * @description Triggers action to delete a user
	 * @param {object} [userId]
	 * @return {void}
	 */
	handleDeleteUser(userId) {
		confirmAlert({
			title: 'Confirm Delete',
			message: 'Are you sure ?',
			confirmLabel: 'Confirm',
			cancelLabel: 'Cancel',
			onConfirm: () => {
				this.props.removeUser(userId).then(() => {
					Materialize.toast('User Deleted Successfully', 4000);
				});
			},
			onCancel: () => ''
		});
	}
	/**
	 * @description Triggers search action
	 * @param {object} e
	 * @return {void}
	 */
	handleSearch(e) {
		e.preventDefault();
		if (this.state.searchQuery !== '') {
			this.props.searchUsers(this.state).then(() => {
			});
		} else {
			this.props.allUsers(this.state);
		}
	}
	/**
	 * @description Render content to the screen
	 * @return {void}
	 */
	render() {
		const users = _.map(this.state.users);

		return (
			<div>
				{
					(this.state.users.length !== 0) ? (
						<ReactPaginate
							previousLabel={
								<i className="fa fa-chevron-left fa-2x" aria-hidden="true" />
							}
							nextLabel={
								<i className="fa fa-chevron-right fa-2x" id="btn-Next" aria-hidden="true" />
							}
							breakLabel={<a href="">...</a>}
							breakClassName={'break-me'}
							pageCount={this.state.pagination.pages}
							initialPage={0}
							marginPagesDisplayed={2}
							pageRangeDisplayed={5}
							onPageChange={this.handlePageChange}
							containerClassName={'user-pagination pagination'}
							subContainerClassName={'pages pagination'}
							activeClassName={'active'}
						/>
					) : ('')
				}
				<div className="user-panel">
					<div className="f-center scroll">
						<h5 className="user-panel-header"><span>All Users</span></h5>
						<table className="striped centered">
							<thead>
								<tr>
									<th>First Name</th>
									<th>Last Name</th>
									<th>Email</th>
									<th>Role #</th>
									<th><div className="user-search">
										<form >
											<input
												type="text"
												id="search"
												name="search"
												value={this.state.searchQuery}
												onChange={this.onChange}
												onKeyUp={this.handleSearch}
												placeholder="Search by name"
											/>
										</form>
									</div></th>
								</tr>
							</thead>
							<tbody>
								{users.map(user => (
									<tr id="users-table">
										<td id="fname">{user.firstName}</td>
										<td>{user.lastName}</td>
										<td>{user.email}</td>
										<td>{user.roleId}</td>
										<td>
											<a
												tabIndex="0"
												role="button"
												id="btn-userdelete"
												onClick={() => this.handleDeleteUser(user.id)}
											>
												<i className="fa fa-trash" aria-hidden="true" />
											</a>
										</td>
									</tr>
								)) }
							</tbody>
						</table>
					</div>
				</div>
			</div>
		);
	}
}
AllUsers.propTypes = {
	allUsers: propTypes.func.isRequired,
	removeUser: propTypes.func.isRequired,
	users: propTypes.shape({}).isRequired,
	searchUsers: propTypes.func.isRequired,
	pagination: propTypes.shape({
		pages: propTypes.number.isRequired
	}).isRequired
	// searchResults: propTypes.shape({
	// 	length: propTypes.number.isRequired
	// }).isRequired
};
/**
 * @description Maps State to Props
 * @param  {object} state
 * @return {void}
 */
function mapStateToProps(state) {
	return {
		users: state.users.allUsers,
		pagination: state.users.pagination,
		searchResult: state.users.userSearch
	};
}
export default withRouter(connect(mapStateToProps,
	{ allUsers, removeUser, searchUsers })(AllUsers));

