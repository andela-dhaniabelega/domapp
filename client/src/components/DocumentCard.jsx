import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import { confirmAlert } from 'react-confirm-alert';
import 'react-confirm-alert/src/react-confirm-alert.css';
import { removeDocument } from '../actions/removeDocument';
import EditDocumentForm from './EditDocumentForm';
/**
 * @class Documents
 * @extends {React.Component}
 */
class DocumentCard extends React.Component {
	/**
	 * @param  {object} props
	 * @return {void}
	 */
	constructor(props) {
		super(props);
		this.openModal = this.openModal.bind(this);
		this.handleDocumentDelete = this.handleDocumentDelete.bind(this);
	}
	/**
	 * @description Modal to Edit a Document
	 * @return {void}
	 */
	openModal() {
		$(`.modal.${this.props.document.id}`).modal({
			dismissible: true, // Modal can be dismissed by clicking outside of the modal
			opacity: 0.5, // Opacity of modal background
		});
		$(`.modal.${this.props.document.id}`).modal();
	}
	/**
	 * @description Opens document details page
	 * @param  {string} link
	 * @return {void}
	 */
	getDocument(link) {
		this.props.history.push(link);
	}
	/**
	 * @description handles document deletion
	 * @param  {integer} documentId
	 * @return {void}
	 */
	handleDocumentDelete(documentId) {
		confirmAlert({
			title: 'Confirm Delete',
			message: 'Are you sure ?',
			// childrenElement: () => <div>Custom UI</div>,
			confirmLabel: 'Confirm',
			cancelLabel: 'Cancel',
			onConfirm: () => {
				this.props.removeDocument(documentId).then(() => {
					Materialize.toast('Document Deleted Successfully', 4000);
				});
			},
			onCancel: () => ''
		});
	}
	/**
	 * @return {void}
	 */
	render() {
		const createdAt = this.props.document.createdAt;
		let content = this.props.document.content;
		const createdDate = createdAt.split('T')[0];
		content = content.length > 167 ? content.substring(0, 167) + '...' : content;
		const documentList = (
			<div className="col s4 m4 darken-1">
				<div className="card">
					<div className="card-content">
						<span className="card-title truncate">
							<a
								onClick={() => this.getDocument(`/document/${this.props.document.id}`)}
								role="button"
							>
								{ this.props.document.title }
							</a>
						</span>
						<span className="created">{createdDate}</span>
						<p className="grey-light">{content}</p>
					</div>
					<div className="card-action form-card-action">
						{
							(this.props.auth.user.id === this.props.document.userId) ?
								(
									<div>
										<a href="#editModal" onClick={this.openModal}>Edit</a>
										<a href="#!" onClick={() => this.handleDocumentDelete(this.props.document.id)}>Delete</a>
									</div>
								) : (<div><i className="fa fa-lock" aria-hidden="true" /></div>)
						}
					</div>
				</div>
			</div>
		);
		return (
			<div>
				{this.props.document.length === 0 ? <div><p>No new docs</p></div> : documentList}
				<div id="editModal" className={`modal ${this.props.document.id}`}>
					<div className="modal-content">
						<h5>Edit Document</h5>
						<EditDocumentForm document={this.props.document} />
					</div>
				</div>
			</div>
		);
	}
}
DocumentCard.propTypes = {
	document: propTypes.object.isRequired,
	removeDocument: propTypes.func.isRequired,
	auth: propTypes.object.isRequired
};
/**
 * @description Maps State to Props
 * @param  {object} state
 * @return {void}
 */
function mapStateToProps(state) {
	return {
		auth: state.auth
	};
}
export default withRouter(connect(mapStateToProps, { removeDocument })(DocumentCard));
