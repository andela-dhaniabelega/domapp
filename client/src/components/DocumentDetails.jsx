import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import propTypes from 'prop-types';
import Parser from 'html-react-parser';
import jwt from 'jwt-decode';
import tinyMceConfig from '../utils/tinyMceConfig.json';
import { documentDetails } from '../actions/documentDetails';
import { editDocument } from '../actions/editDocument';
import validateInput from '../../../server/helpers/editDocumentValidation';
/**
 * @class Documents
 * @extends {React.Component}
 */
export class DocumentDetails extends React.Component {
	/**
	 * @param  {object} props
	 * @return {void}
	 */
	constructor(props) {
		super(props);
		// this.decoded = jwt(localStorage.getItem('token'));
		this.state = {
			id: this.props.match.params.id,
			title: '',
			access: '',
			content: '',
			edit: false,
			errors: {},
			documents: [],
			decoded: jwt(localStorage.getItem('token'))
		};
		this.openEditor = this.openEditor.bind(this);
		this.closeEditor = this.closeEditor.bind(this);
		this.updateDocument = this.updateDocument.bind(this);
		this.onChange = this.onChange.bind(this);
	}
	/**
	 * @description Lifcycle Method
	 * @return {void}
	 */
	componentDidMount() {
		this.props.documentDetails(this.state.id).then(() => {
			this.setState({
				title: this.props.document.title,
				content: this.props.document.content,
				access: this.props.document.access
			});
		});
	}
	/**
	 * @description Lifecycle Method
	 * @param  {object} nextProps
	 * @return {void}
	 */
	componentWillReceiveProps(nextProps) {
		this.setState({ documents: nextProps.document });
	}
	/**
	 * @param  {object} e
	 * @return {void}
	 */
	onChange(e) {
		this.setState({ [e.target.name]: e.target.value });
	}
	/**
	 * @description Checks for form validity
	 * @return {Boolean}
	 */
	validateForm() {
		const { errors, isValid } = validateInput(this.state);
		if (!isValid) {
			this.setState({ errors });
		}
		return isValid;
	}
	/**
	 * @description Activates TinyMCE Editor
	 * @return {void}
	 */
	openEditor() {
		this.setState({ edit: true });
		tinymce.init(tinyMceConfig);
	}
	/**
	 * @description De-Activates TinyMCE Editor
	 * @return {void}
	 */
	closeEditor() {
		tinymce.remove('#e-content');
		this.setState({
			edit: false,
			title: this.props.document.title,
			content: this.props.document.content,
			access: this.props.document.access
		});
	}
	/**
	 * @description Triggers action to update document
	 * @return {void}
	 */
	updateDocument() {
		if (this.validateForm()) {
			this.setState({ errors: {} });
			const content = tinymce.activeEditor.getContent();
			const parsedContent = Parser(content);
			const newDocument = {
				id: this.state.id,
				title: this.state.title,
				content: parsedContent.props.children,
				access: this.state.access
			};
			this.props.editDocument(newDocument).then(() => {
				tinymce.remove('#e-content');
				this.setState({ edit: false });
				Materialize.toast('Document Updated Successfully', 4000);
				this.setState({
					title: this.props.document.title,
					content: this.props.document.content,
					access: this.props.document.access
				});
			}).catch((err) => {
				Materialize.toast(err.response.data.message, 4000);
			});
		}
	}
	/**
	 * @description Render content to the screen
	 * @return {void}
	 */
	render() {
		let errors = {};
		if (this.state.errors !== null) {
			errors = this.state.errors;
		}
		const userId = this.state.decoded.id;
		const documentId = this.state.documents.userId;
		const accessName = ['Public', 'Private', 'Role'];
		const accessOptions = accessName.map(value => (
			<option key={value} value={value}> {value} </option>
		));
		return (
			<div className="doc-wrapper">
				<div className="document-panel">
					<div className="f-center">
						<h5 className="document-panel-header"><span>Document Details</span></h5><br /><br />
						{this.state.edit ? (
							<div>
								<span className="edit-doc-error">{errors.title}</span><br />
								<span className="edit-doc-error">{errors.content}</span><br />
							</div>
						) : ''}
						<div>
							{ userId === documentId ?
								(
									<div className="create-doc">
										<a href="#!" className="create-doc-link" id="btn-editDoc" onClick={this.openEditor}>Edit</a>
									</div>) : ''}
						</div>
						<div>
							{
								this.state.edit ? (
									<div className="edit-panel">
										<input
											type="text"
											id="title"
											name="title"
											value={this.state.title}
											onChange={this.onChange}
										/>
										<select
											name="access"
											id="access"
											className="browser-default"
											value={this.state.access}
											onChange={this.onChange}
										>
											{ accessOptions }
										</select>
									</div>
								) : (
									<div className="title-content-panel">
										<div className="editor-title-header">
											Title: <div className="editor-title">{this.state.title}</div>
										</div>
										<div className="access-header-title">
											Access: <div className="editor-access">{this.state.access}</div>
										</div>
									</div>
								)
							}
						</div>
						<div className="editor-container">
							<div className="editor-content" id="e-content">
								{ Parser(this.state.content) }
							</div>
						</div>
					</div>
					{ (this.state.edit) ?
						(<div className="editor-buttons">
							<button
								onClick={this.updateDocument}
								className="button-primary button-block"
								id="btn-updateDocument"
							>Save</button>
							<button
								onClick={this.closeEditor}
								className="button-primary button-block"
								id="btn-closeEditor"
							>
								Cancel
							</button>
						</div>
						) : ''}
				</div>
			</div>
		);
	}
}
DocumentDetails.propTypes = {
	document: propTypes.shape({
		title: propTypes.string.isRequired,
		content: propTypes.string.isRequired,
		access: propTypes.string.isRequired,
		userId: propTypes.number.isRequired
	}).isRequired,
	documentDetails: propTypes.func.isRequired,
	editDocument: propTypes.func.isRequired,
	match: propTypes.shape({
		params: propTypes.shape({
			id: propTypes.string
		})
	}).isRequired
};
/**
 * @description Maps State to Props
 * @param  {object} state
 * @return {void}
 */
function mapStateToProps(state) {
	return {
		document: state.userDocuments.document,
	};
}
export default withRouter(connect(mapStateToProps,
	{ documentDetails, editDocument })(DocumentDetails));

