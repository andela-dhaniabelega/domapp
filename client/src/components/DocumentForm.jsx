import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import propTypes from 'prop-types';
import TinyMCE from 'react-tinymce';
import classnames from 'classnames';
import { newDocument } from '../actions/newDocument';
import { allDocuments } from '../actions/allDocuments';
import createDocumentValidation from '../../../server/helpers/createDocumentValidation';

/**
 * 
 * @class SignInForm
 * @extends {React.Component}
 */
export class DocumentForm extends React.Component {
	/**
	 * @description cerates and intializes objects
	 * @param  {object} props
	 * @return {void}
	 */
	constructor(props) {
		super(props);
		this.state = {
			btnStatus: false,
			title: '',
			content: '',
			access: '',
			errors: '',
			limit: this.props.limit,
			offset: this.props.offset,
			success: {}
		};
		this.onChange = this.onChange.bind(this);
		this.createDocument = this.createDocument.bind(this);
		this.onEditorChange = this.onEditorChange.bind(this);
	}
	/**
	 * @description Gets content from TinyMCE ediotr
	 * @param  {object} e
	 * @return {void}
	 */
	onEditorChange(e) {
		this.setState({ content: e.target.getContent() });
	}
	/**
	 * @description Allows user Interact with Input boxes
	 * @param  {object} e
	 * @return {void}
	 */
	onChange(e) {
		this.setState({ [e.target.name]: e.target.value });
	}
	/**
	 * @description Handles Document Creation
	 * @return {void}
	 */
	createDocument() {
		if (this.validateForm()) {
			this.setState({
				errors: {},
				title: '',
				content: tinymce.activeEditor.setContent(''),
				access: '' });
			this.props.newDocument(this.state).then(() => {
				this.props.allDocuments(this.state);
				this.props.closeModal('close');
				Materialize.toast('Document Created', 4000);
			}).catch((err) => {
				Materialize.toast(err.response.data.message, 4000);
				this.props.closeModal('close');
			});
		} else {
			this.setState({ btnStatus: false });
		}
	}
	/**
	 * @description Checks for form validity
	 * @return {Boolean}
	 */
	validateForm() {
		const { errors, isValid } = createDocumentValidation(this.state);
		if (!isValid) {
			this.setState({ errors });
		}
		return isValid;
	}
	/**
	 * @description Renders content to the screen
	 * @return {void}
	 */
	render() {
		let errors = {};
		if (this.state.errors !== null) {
			errors = this.state.errors;
		}
		console.log(errors, 'the errors');
		const modalBtn = classnames({
			'button-primary': true,
			'button-block': true,
			'modal-close': this.state.btnStatus
		});
		return (
			<div>
				<div className="form">
					<div className="f-center">
						<h5>New Document</h5> <br />
						<span className="sign-up-error">{errors.title}</span><br />
						<span className="sign-up-error">{errors.content}</span><br />
						<label htmlFor="title">Title</label>
						<input
							className="browser-defaults"
							type="text"
							name="title"
							id="title"
							value={this.state.title}
							onChange={this.onChange}
							placeholder="Title"
						/>
						<br />
						<label htmlFor="selectbox">Access</label>
						<select name="access" onChange={this.onChange}>
							<option value="public">Public</option>
							<option value="private">Private</option>
							<option value="role">Role</option>
						</select>
						<br />
						<div id="tiny-edit">
							<TinyMCE
								config={{
									plugins: 'link image code',
									toolbar: 'undo redo | bold italic | alignleft aligncenter alignright | code'
								}}
								content={this.state.content}
								onChange={this.onEditorChange}
							/>
						</div>
						<br />
						<button
							className={modalBtn}
							id="btn-createdocument"
							onClick={this.createDocument}
						>
							Create
						</button>
					</div>
				</div>
			</div>
		);
	}
}
DocumentForm.propTypes = {
	newDocument: propTypes.func.isRequired,
	allDocuments: propTypes.func.isRequired,
	closeModal: propTypes.func.isRequired,
	limit: propTypes.number.isRequired,
	offset: propTypes.number.isRequired
};

/**
 * @description Maps State to Props
 * @param  {object} state
 * @return {object}
 */
function mapStateToProps(state) {
	return {
		userDocuments: state.documents,
	};
}

export default withRouter(connect(mapStateToProps, { newDocument, allDocuments })(DocumentForm));
