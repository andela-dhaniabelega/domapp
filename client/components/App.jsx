import React from 'react';
import Header from './Header';

// import {connect} from 'react-redux';

// const mapStateToProps = state => ({appName: state.appName});

class App extends React.Component {
  render() {
    return (
      <div >
        <Header /> 
        {this.props.children}
      </div>
    );
  }
}
export default App;

// export default connect(mapStateToProps, () => ({}))(App);

