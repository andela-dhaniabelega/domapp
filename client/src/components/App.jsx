import React from 'react';
import PropTypes from 'prop-types';
import Header from './Header';

/**
 * @description Container for all components
 * @return {void}
 */
const App = () => ({
	render() {
		return (
			<div className="main">
				<Header />
				{this.props.children}
				<div className="clear" />
			</div>
		);
	}
});

App.propTypes = {
	children: PropTypes.element.isRequired
};
export default App;

