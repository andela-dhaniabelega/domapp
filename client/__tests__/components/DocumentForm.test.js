import React from 'react';
import { assert } from 'chai';
import { spy } from 'sinon';
import { mount, shallow } from 'enzyme';
import { DocumentForm } from '../../src/components/DocumentForm';

const newDocumentMock = jest.fn();
const allDocumentsMock = () => {};
const closeModal = jest.fn();

describe('Document Form', () => {
	it('renders a snapshot', () => {
		const wrapper = shallow(<DocumentForm />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should have an onChange function', () => {
		const onChangeSpy = spy(DocumentForm.prototype, 'onChange');
		const event = { target: { name: 'title', value: 'hoyt' } };
		const wrap = mount(
			<DocumentForm />
		);
		wrap.find('input#title').simulate('change', onChangeSpy, event);
		assert.ok(onChangeSpy.calledOnce);
	});
	it('should have a createDocument() function', () => {
		const wrapper = mount(
			<DocumentForm
				newDocument={newDocumentMock}
				allDocuments={allDocumentsMock}
				closeModal={closeModal}
			/>
		);
		const button = wrapper.find('#btn-createdocument');
		button.simulate('click');
		assert.equal(button.length, 1);
	});
});
