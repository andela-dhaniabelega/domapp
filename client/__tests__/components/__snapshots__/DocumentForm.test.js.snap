// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Form renders a snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
        className="form"
    >
        <div
            className="f-center"
        >
            <h5>
                New Document
            </h5>
             
            <br />
            <span
                className="sign-up-error"
            />
            <br />
            <span
                className="sign-up-error"
            />
            <br />
            <label
                htmlFor="title"
            >
                Title
            </label>
            <input
                className="browser-defaults"
                id="title"
                name="title"
                onChange={[Function]}
                placeholder="Title"
                type="text"
                value=""
            />
            <br />
            <label
                htmlFor="selectbox"
            >
                Access
            </label>
            <select
                name="access"
                onChange={[Function]}
            >
                <option
                    value="public"
                >
                    Public
                </option>
                <option
                    value="private"
                >
                    Private
                </option>
                <option
                    value="role"
                >
                    Role
                </option>
            </select>
            <br />
            <div>
                <TinyMCE
                    config={
                        Object {
                            "plugins": "link image code",
                            "toolbar": "undo redo | bold italic | alignleft aligncenter alignright | code",
                          }
                    }
                    content=""
                    onChange={[Function]}
                />
            </div>
            <br />
            <button
                className="button-primary button-block"
                id="btn-createdocument"
                onClick={[Function]}
            >
                Create
            </button>
        </div>
    </div>
</div>,
  "nodes": Array [
    <div>
      <div
            className="form"
      >
            <div
                  className="f-center"
            >
                  <h5>
                        New Document
                  </h5>
                   
                  <br />
                  <span
                        className="sign-up-error"
                  />
                  <br />
                  <span
                        className="sign-up-error"
                  />
                  <br />
                  <label
                        htmlFor="title"
                  >
                        Title
                  </label>
                  <input
                        className="browser-defaults"
                        id="title"
                        name="title"
                        onChange={[Function]}
                        placeholder="Title"
                        type="text"
                        value=""
                  />
                  <br />
                  <label
                        htmlFor="selectbox"
                  >
                        Access
                  </label>
                  <select
                        name="access"
                        onChange={[Function]}
                  >
                        <option
                              value="public"
                        >
                              Public
                        </option>
                        <option
                              value="private"
                        >
                              Private
                        </option>
                        <option
                              value="role"
                        >
                              Role
                        </option>
                  </select>
                  <br />
                  <div>
                        <TinyMCE
                              config={
                                    Object {
                                          "plugins": "link image code",
                                          "toolbar": "undo redo | bold italic | alignleft aligncenter alignright | code",
                                        }
                              }
                              content=""
                              onChange={[Function]}
                        />
                  </div>
                  <br />
                  <button
                        className="button-primary button-block"
                        id="btn-createdocument"
                        onClick={[Function]}
                  >
                        Create
                  </button>
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DocumentForm />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DocumentForm {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "createDocument": [Function],
        "onChange": [Function],
        "onEditorChange": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "access": "",
          "btnStatus": false,
          "content": "",
          "errors": "",
          "limit": undefined,
          "offset": undefined,
          "success": Object {},
          "title": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
                    className="form"
          >
                    <div
                              className="f-center"
                    >
                              <h5>
                                        New Document
                              </h5>
                               
                              <br />
                              <span
                                        className="sign-up-error"
                              />
                              <br />
                              <span
                                        className="sign-up-error"
                              />
                              <br />
                              <label
                                        htmlFor="title"
                              >
                                        Title
                              </label>
                              <input
                                        className="browser-defaults"
                                        id="title"
                                        name="title"
                                        onChange={[Function]}
                                        placeholder="Title"
                                        type="text"
                                        value=""
                              />
                              <br />
                              <label
                                        htmlFor="selectbox"
                              >
                                        Access
                              </label>
                              <select
                                        name="access"
                                        onChange={[Function]}
                              >
                                        <option
                                                  value="public"
                                        >
                                                  Public
                                        </option>
                                        <option
                                                  value="private"
                                        >
                                                  Private
                                        </option>
                                        <option
                                                  value="role"
                                        >
                                                  Role
                                        </option>
                              </select>
                              <br />
                              <div>
                                        <TinyMCE
                                                  config={
                                                            Object {
                                                                      "plugins": "link image code",
                                                                      "toolbar": "undo redo | bold italic | alignleft aligncenter alignright | code",
                                                                    }
                                                  }
                                                  content=""
                                                  onChange={[Function]}
                                        />
                              </div>
                              <br />
                              <button
                                        className="button-primary button-block"
                                        id="btn-createdocument"
                                        onClick={[Function]}
                              >
                                        Create
                              </button>
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
                    className="form"
          >
                    <div
                              className="f-center"
                    >
                              <h5>
                                        New Document
                              </h5>
                               
                              <br />
                              <span
                                        className="sign-up-error"
                              />
                              <br />
                              <span
                                        className="sign-up-error"
                              />
                              <br />
                              <label
                                        htmlFor="title"
                              >
                                        Title
                              </label>
                              <input
                                        className="browser-defaults"
                                        id="title"
                                        name="title"
                                        onChange={[Function]}
                                        placeholder="Title"
                                        type="text"
                                        value=""
                              />
                              <br />
                              <label
                                        htmlFor="selectbox"
                              >
                                        Access
                              </label>
                              <select
                                        name="access"
                                        onChange={[Function]}
                              >
                                        <option
                                                  value="public"
                                        >
                                                  Public
                                        </option>
                                        <option
                                                  value="private"
                                        >
                                                  Private
                                        </option>
                                        <option
                                                  value="role"
                                        >
                                                  Role
                                        </option>
                              </select>
                              <br />
                              <div>
                                        <TinyMCE
                                                  config={
                                                            Object {
                                                                      "plugins": "link image code",
                                                                      "toolbar": "undo redo | bold italic | alignleft aligncenter alignright | code",
                                                                    }
                                                  }
                                                  content=""
                                                  onChange={[Function]}
                                        />
                              </div>
                              <br />
                              <button
                                        className="button-primary button-block"
                                        id="btn-createdocument"
                                        onClick={[Function]}
                              >
                                        Create
                              </button>
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DocumentForm />,
}
`;
